### PR TITLE
Unify sparql commands

### PR DESCRIPTION
**Pull request by dougli1sqrd**

### CODE REVIEW

**Comment by jamesaoverton** [comment]
Is there any documentation?

**Comment by jamesaoverton** [comment]
I'd like some sort of textual description of what this is supposed to be doing, so that I can tell whether it's doing what it should.

**Comment by dougli1sqrd** [comment]
That's true, I'll include the options in the readme. Also I'll fix the conflicts.

**Comment by jamesaoverton** [comment]
@dougli1sqrd I would really like to move forward with this feature.

**Comment by cmungall** [comment]
We'll get on this today

**Comment by jamesaoverton** [comment]
Detecting the SPARQL type is a good idea. The current code for `--sparql` only allows one query and one output, while previous versions allowed for loading an ontology once and running many queries. That seems like a valuable feature to me.

**Comment by dougli1sqrd** [comment]
oh really? In the normal way, how would you run multiple queries on an ontology?

**Comment by jamesaoverton** [comment]
It’s not pretty, but these loops do the job: https://github.com/ontodev/robot/blob/master/robot-command/src/main/java/org/obolibrary/robot/QueryCommand.java#L152

File: examples/README.md [hunk 1]
```diff
@@ -131,6 +131,17 @@ Has many results, but a portion could look like:
             
 Note that we've created a triple where none existed before. 
 
+## Alternate Query Form with `--sparql`
+
+For any non-updating query, you don't have to use the above forms, where you need to know the kind of query 
+before you run it. Instead, you can simply supply `--sparql` and then the path to the sparql query file, `--output` and
```
Comment (jamesaoverton) [inline examples/README.md]: I think this changed...

File: examples/basic.sparql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+SELECT *
```
Comment (jamesaoverton) [inline examples/basic.sparql]: Is this file used?

File: robot-command/src/main/java/org/obolibrary/robot/QueryCommand.java [hunk 3]
```diff
@@ -121,49 +129,83 @@ public CommandState execute(CommandState state, String[] args)
...
+//            String query = FileUtils.readFileToString(new File(queryPath));
+//
+//            QueryOperation.runSparqlQuery(dsg, query, new File(outputPath), outputFormat);
+//            return state;
+//        }
+
+        if(line.hasOption("sparql")) {
+            List<String> sparql = CommandLineHelper.getOptionalValues(line, "sparql");
+
+            for (int i=0; i<sparql.size(); i += 2) {
```
Comment (jamesaoverton) [inline robot-command/src/main/java/org/obolibrary/robot/QueryCommand.java]: Thanks

File: robot-command/src/main/java/org/obolibrary/robot/QueryCommand.java [hunk 4]
```diff
@@ -121,49 +129,83 @@ public CommandState execute(CommandState state, String[] args)
...
             }
         }
 
         IOHelper ioHelper = CommandLineHelper.getIOHelper(line);
         state = CommandLineHelper.updateInputOntology(ioHelper, state, line);
         DatasetGraph dsg = QueryOperation.loadOntology(state.getOntology());
 
-        // Handle select
-        if (line.hasOption("select")) {
+//        if(line.hasOption("sparql") && line.hasOption("output")) {
```
Comment (jamesaoverton) [inline robot-command/src/main/java/org/obolibrary/robot/QueryCommand.java]: Please clean up all the commented lines.

**Pull request merged**